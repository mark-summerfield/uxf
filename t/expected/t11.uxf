uxf 1.0 TLM Config
= Categories CID Title Selected
= Files Kind Filename
= IPv4 A:int B:int C:int D:int
= Playlists PID Title CID Selected
= RGB Red:int Green:int Blue:int
= RGBA Red:int Green:int Blue:int Alpha:int
{
  <General> {
    <saved> 2022-03-21
    <autosave> yes
    <historysize> -35
    <volume> 0.7
    <files> (Files
      <current> </home/mark/app/rs/tlm/PlaylistsTest.tlm>
      <recent1> </home/mark/app/rs/tlm/PlaylistsTest.tlm>
      <recent2> </home/mark/data/playlists-all.tlm>
    )
  }
  <Window> {
    <pos> [383 124]
    <size> [590 536]
    <scale> 1.1
  }
  <More Tests> [ #<A list of name value pairs; normally would be a map> <one> (IPv4 141 179 19 9) <many> (IPv4
  172 16 254 1
  192 0 2 235
  127 0 0 1
 )
 <red> (RGB 255 0 0) <semi-transparent blue> (RGBA 0 0 255 80)]
  <Magic> (:1F8B:)
  <Nested Map> {
    <Classical> [5 yes]
    <Modern Instrumental> [4 no]
    <New Acquistions> [1 no]
    <Nested List of Lists> [
      [5 <Classical> yes]
      [4 <Modern Instrumental> no]
      [1 <New Acquistions> no]
    ]
    <Nested Table> (Categories
      5 <Classical> yes
      4 <Modern Instrumental> no
      1 <New Acquistions> no
      2 <Pop> no
      3 <Punk> no
      7 <Uncategorized> no
      6 <Unpopular Pop> no
    )
    <Nested List of Tables> [
      (Categories
        5 <Classical> yes
        4 <Modern Instrumental> no
        1 <New Acquistions> no
        2 <Pop> no
        3 <Punk> no
        7 <Uncategorized> no
        6 <Unpopular Pop> no
      )
      (Playlists
        4 <ABBA> 2 no
        38 <Bach> 5 no
        39 <Bartok> 5 no
        5 <Beatles> 2 no
        40 <Beethoven> 5 no
        6 <Blondie> 2 no
        52 <Bob Marley> 6 yes
        7 <Bruce Springsteen> 2 no
        41 <Chopin> 5 yes
        37 <Classical> 5 no
        8 <David Bowie> 2 no
        9 <Dire Straits> 2 no
      )
    ]
  }
}
