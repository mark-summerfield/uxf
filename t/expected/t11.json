{
  "UXF^custom": "TLM Config",
  "UXF^ttypes": [
    {
      "UXF^ttype": {
        "name": "Files",
        "fields": {
          "Kind": null,
          "Filename": null
        }
      }
    },
    {
      "UXF^ttype": {
        "name": "Categories",
        "fields": {
          "CID": null,
          "Title": null,
          "Selected": null
        }
      }
    },
    {
      "UXF^ttype": {
        "name": "Playlists",
        "fields": {
          "PID": null,
          "Title": null,
          "CID": null,
          "Selected": null
        }
      }
    },
    {
      "UXF^ttype": {
        "name": "IPv4",
        "fields": {
          "A": "int",
          "B": "int",
          "C": "int",
          "D": "int"
        }
      }
    },
    {
      "UXF^ttype": {
        "name": "RGB",
        "fields": {
          "Red": "int",
          "Green": "int",
          "Blue": "int"
        }
      }
    },
    {
      "UXF^ttype": {
        "name": "RGBA",
        "fields": {
          "Red": "int",
          "Green": "int",
          "Blue": "int",
          "Alpha": "int"
        }
      }
    }
  ],
  "UXF^data": {
    "General": {
      "saved": {
        "UXF^date": "2022-03-21"
      },
      "autosave": true,
      "historysize": -35,
      "volume": 0.7,
      "files": {
        "UXF^table": {
          "name": "Files",
          "comment": null,
          "fields": {
            "Kind": null,
            "Filename": null
          },
          "records": [
            [
              "current",
              "/home/mark/app/rs/tlm/PlaylistsTest.tlm"
            ],
            [
              "recent1",
              "/home/mark/app/rs/tlm/PlaylistsTest.tlm"
            ],
            [
              "recent2",
              "/home/mark/data/playlists-all.tlm"
            ]
          ]
        }
      }
    },
    "Window": {
      "pos": [
        383,
        124
      ],
      "size": [
        590,
        536
      ],
      "scale": 1.1
    },
    "More Tests": {
      "UXF^list": {
        "comment": "A list of name value pairs; normally would be a map",
        "vtype": null,
        "list": [
          "one",
          {
            "UXF^table": {
              "name": "IPv4",
              "comment": null,
              "fields": {
                "A": "int",
                "B": "int",
                "C": "int",
                "D": "int"
              },
              "records": [
                [
                  141,
                  179,
                  19,
                  9
                ]
              ]
            }
          },
          "many",
          {
            "UXF^table": {
              "name": "IPv4",
              "comment": null,
              "fields": {
                "A": "int",
                "B": "int",
                "C": "int",
                "D": "int"
              },
              "records": [
                [
                  172,
                  16,
                  254,
                  1
                ],
                [
                  192,
                  0,
                  2,
                  235
                ],
                [
                  127,
                  0,
                  0,
                  1
                ]
              ]
            }
          },
          "red",
          {
            "UXF^table": {
              "name": "RGB",
              "comment": null,
              "fields": {
                "Red": "int",
                "Green": "int",
                "Blue": "int"
              },
              "records": [
                [
                  255,
                  0,
                  0
                ]
              ]
            }
          },
          "semi-transparent blue",
          {
            "UXF^table": {
              "name": "RGBA",
              "comment": null,
              "fields": {
                "Red": "int",
                "Green": "int",
                "Blue": "int",
                "Alpha": "int"
              },
              "records": [
                [
                  0,
                  0,
                  255,
                  80
                ]
              ]
            }
          }
        ]
      }
    },
    "Magic": {
      "UXF^bytes": "1F8B"
    },
    "Nested Map": {
      "Classical": [
        5,
        true
      ],
      "Modern Instrumental": [
        4,
        false
      ],
      "New Acquistions": [
        1,
        false
      ],
      "Nested List of Lists": [
        [
          5,
          "Classical",
          true
        ],
        [
          4,
          "Modern Instrumental",
          false
        ],
        [
          1,
          "New Acquistions",
          false
        ]
      ],
      "Nested Table": {
        "UXF^table": {
          "name": "Categories",
          "comment": null,
          "fields": {
            "CID": null,
            "Title": null,
            "Selected": null
          },
          "records": [
            [
              5,
              "Classical",
              true
            ],
            [
              4,
              "Modern Instrumental",
              false
            ],
            [
              1,
              "New Acquistions",
              false
            ],
            [
              2,
              "Pop",
              false
            ],
            [
              3,
              "Punk",
              false
            ],
            [
              7,
              "Uncategorized",
              false
            ],
            [
              6,
              "Unpopular Pop",
              false
            ]
          ]
        }
      },
      "Nested List of Tables": [
        {
          "UXF^table": {
            "name": "Categories",
            "comment": null,
            "fields": {
              "CID": null,
              "Title": null,
              "Selected": null
            },
            "records": [
              [
                5,
                "Classical",
                true
              ],
              [
                4,
                "Modern Instrumental",
                false
              ],
              [
                1,
                "New Acquistions",
                false
              ],
              [
                2,
                "Pop",
                false
              ],
              [
                3,
                "Punk",
                false
              ],
              [
                7,
                "Uncategorized",
                false
              ],
              [
                6,
                "Unpopular Pop",
                false
              ]
            ]
          }
        },
        {
          "UXF^table": {
            "name": "Playlists",
            "comment": null,
            "fields": {
              "PID": null,
              "Title": null,
              "CID": null,
              "Selected": null
            },
            "records": [
              [
                4,
                "ABBA",
                2,
                false
              ],
              [
                38,
                "Bach",
                5,
                false
              ],
              [
                39,
                "Bartok",
                5,
                false
              ],
              [
                5,
                "Beatles",
                2,
                false
              ],
              [
                40,
                "Beethoven",
                5,
                false
              ],
              [
                6,
                "Blondie",
                2,
                false
              ],
              [
                52,
                "Bob Marley",
                6,
                true
              ],
              [
                7,
                "Bruce Springsteen",
                2,
                false
              ],
              [
                41,
                "Chopin",
                5,
                true
              ],
              [
                37,
                "Classical",
                5,
                false
              ],
              [
                8,
                "David Bowie",
                2,
                false
              ],
              [
                9,
                "Dire Straits",
                2,
                false
              ]
            ]
          }
        }
      ]
    }
  }
}